crates/bevy_ui/src/render/mod.rs

```diff
@@ -618,13 +619,12 @@ pub fn extract_text_uinodes(
     >,
 ) {
     // TODO: Support window-independent UI scale: https://github.com/bevyengine/bevy/issues/5621
-    let scale_factor = windows
-        .get_single()
-        .map(|window| window.resolution.scale_factor())
-        .unwrap_or(1.0)
-        * ui_scale.0;
 
-    let inverse_scale_factor = (scale_factor as f32).recip();
+    let scale_factor = windows.get_single().map_or(1., |window| {
+        (window.resolution.scale_factor() * ui_scale.0) as f32
```

Comment:
This applies ui_scale only if the window exists. Shouldn't ui_scale be used no matter what?